src/vaadin-grid-filter-mixin.js

```diff
@@ -29,13 +29,38 @@ export const FilterMixin = (superClass) =>
 
     /** @private */
     _filterChanged(e) {
-      if (this._filters.indexOf(e.target) === -1) {
-        this._filters.push(e.target);
+      e.stopPropagation();
+
+      this.__updateFilter(e.target);
+      this.__applyFilters();
+    }
+
+    __removeFilters(filters) {
+      if (filters.length == 0) {
+        return;
       }
 
-      e.stopPropagation();
+      this._filters.forEach((filter) => {
```

Comment:
This looks like a bug: The `filters` param of the method is not used in this loop. Instead it removes every single filter in `this._filters`. It looks like you want to loop on `filters` instead of `this._filters`.

Also wondering if it would not be possible to use `Array.filter` here? Basically:
```
this._filters = this._filters.filter(filter => filters.indexOf(filter) >= 0);
```

Might also be more clear to rename the param to `filtersToRemove`.

Reply:
Done